src/model/state.ts

```diff
@@ -148,14 +153,33 @@ export class State {
     this.snapshot.CurrentIp[EthAddress] = getIpFromHex(event.returnValues.ip);
   }
 
-  applyNewImageVersion(time: number, rolloutGroup: string, imageName: string, imageVersion: string) {
-    if (!Versioning.isValid(imageVersion)) return;
-    this.snapshot.CurrentImageVersionsUpdateTime[rolloutGroup][imageName] = time;
-    const currentVersion = this.snapshot.CurrentImageVersions[rolloutGroup][imageName];
-    // image version upgrades only go forward (we don't allow downgrade)
-    if (!currentVersion || Versioning.compare(imageVersion, currentVersion) > 0) {
-      this.snapshot.CurrentImageVersions[rolloutGroup][imageName] = imageVersion;
-    }
+  applyNewImageVersion(rolloutGroup: string, imageName: string, imageVersion: string) {
+    this.snapshot.CurrentImageVersions[rolloutGroup][imageName] = imageVersion;
+  }
+
+  applyNewImageVersionPollTime(time: number, rolloutGroup: string, imageName: string) {
```

Comment:
This is on purpose.. I wanted the parameter order to match `applyNewProtocolVersionChanged` and `applyNewSubscriptionChanged` that all receive the current time as first argument. `applyNewImageVersionPendingUpdate` is unique because the time there is not current time, but future time of the upgrade. It also had be in the end due to defaults..